src/lib.rs

```diff
@@ -36,6 +41,46 @@ pub static SVM_HOME: Lazy<PathBuf> = Lazy::new(|| {
     }
 });
 
+// Installer type that copies binary data to the appropriate solc binary file:
+// 1. create target file to copy binary data
+// 2. copy data
+// 3. lock file is removed when installer goes out of scope
+struct Installer {
+    // version of solc
+    version: Version,
+    // binary data of the solc executable
+    binbytes: Vec<u8>,
+    // optional lock to pause other threads from proceeding with installation of the same version
+    lock: Option<PathBuf>,
+}
+
+impl Installer {
+    fn install(&self) -> Result<(), SolcVmError> {
+        let version_path = version_path(self.version.to_string().as_str());
+        let solc_path = version_path.join(&format!("solc-{}", self.version));
+        // create solc file.
+        let mut f = fs::File::create(&solc_path)?;
+
+        #[cfg(target_family = "unix")]
+        f.set_permissions(Permissions::from_mode(0o777))?;
+
+        // copy contents over
+        let mut content = Cursor::new(&self.binbytes);
+        std::io::copy(&mut content, &mut f)?;
+
+        Ok(())
+    }
+}
+
+impl Drop for Installer {
+    fn drop(&mut self) {
+        if self.lock.is_some() {
+            fs::remove_file(self.lock.as_ref().unwrap().as_path())
+                .map_or_else(|e| panic!("{}", e.to_string()), |_| ());
```

Comment:
don't think we need to panic here, and just skip if we fail to remove the file?

Reply:
In that case the lock file stays there. And subsequent installs will keep timing out.

I think I'll add a tracing event instead and ignore the failure. Wdyt?